netlib/http/message.py

```diff
@@ -52,7 +52,22 @@ def from_state(cls, state):
         return cls(**state)
 
 
+class CachedDecode(object):
+    __slots__ = ["encoded", "encoding", "decoded"]
+
+    def __init__(self, object, encoding, decoded):
+        self.encoded = object
+        self.encoding = encoding
+        self.decoded = decoded
+
```

Comment:
The cache needs to be bounded in some way. This data can be huge. We should extend LRUCache to do what we want here, and bound it to X megabytes, for some reasonable X.